src/wp-includes/class-wp-user-query.php

```diff
@@ -816,28 +816,41 @@ public function query() {
 				{$this->query_orderby}
 				{$this->query_limit}
 			";
-
-			if ( is_array( $qv['fields'] ) ) {
-				$this->results = $wpdb->get_results( $this->request );
+			$cache_key     = $this->generate_cache_key( $qv, $this->request );
+			$cache_value   = wp_cache_get( $cache_key, 'users' );
+			if ( false !== $cache_value ) {
```

Comment:
```suggestion
			if ( is_array( $cache_value ) ) {
```

How about is_array check then?